lib/cloud_controller/metrics/prometheus_updater.rb

```diff
@@ -62,70 +78,46 @@ def update_thread_info(thread_info)
       update_gauge_metric(:cc_thread_info_event_machine_resultqueue_num_waiting, thread_info[:event_machine][:resultqueue][:num_waiting], 'EventMachine requests waiting in queue')
     end
 
-    def update_failed_job_count(failed_jobs_by_queue, total)
+    def update_failed_job_count(failed_jobs_by_queue)
       failed_jobs_by_queue.each do |key, value|
-        metric_key = :"cc_failed_job_count_#{key.to_s.underscore}"
-        update_gauge_metric(metric_key, value, "Failed jobs for worker #{key}")
+        update_gauge_metric(:cc_failed_jobs_total, value, "Failed jobs for worker #{key}", labels: { queue: key.to_s.underscore })
       end
-
-      update_gauge_metric(:cc_failed_job_count_total, total, 'Total failed jobs')
     end
 
     def update_vitals(vitals)
       vitals.each do |key, value|
+        next if key.to_s.underscore == 'cpu'
+
         metric_key = :"cc_vitals_#{key.to_s.underscore}"
         update_gauge_metric(metric_key, value, "CloudController Vitals: #{key}")
       end
     end
 
-    def update_log_counts(counts)
-      counts.each do |key, value|
-        metric_key = :"cc_log_count_#{key.to_s.underscore}"
-        update_gauge_metric(metric_key, value, "Log count for log level '#{key}'")
-      end
-    end
-
-    def update_task_stats(total_running_tasks, total_memory_in_mb)
-      update_gauge_metric(:cc_tasks_running_count, total_running_tasks, 'Total running tasks')
-      update_gauge_metric(:cc_tasks_running_memory_in_mb, total_memory_in_mb, 'Total memory consumed by running tasks')
-    end
-
-    def update_synced_invalid_lrps(lrp_count)
-      update_gauge_metric(:cc_diego_sync_invalid_desired_lrps, lrp_count, 'Invalid Desired LRPs')
+    def update_task_stats(total_running_tasks, total_memory_in_bytes)
+      update_gauge_metric(:cc_running_tasks_total, total_running_tasks, 'Total running tasks')
+      update_gauge_metric(:cc_running_tasks_memory_bytes, total_memory_in_bytes, 'Total memory consumed by running tasks')
     end
 
     def start_staging_request_received
-      increment_counter_metric(:cc_staging_requested, 'Number of staging requests')
+      increment_counter_metric(:cc_staging_requested_total, 'Number of staging requests')
     end
 
     def report_staging_success_metrics(duration_ns)
-      increment_counter_metric(:cc_staging_succeeded, 'Number of successful staging events')
```

Comment:
Histogram metrics also emit one time series for `_count` and a bucket `+Inf`, which both show the total number. There is no need for another counter metric.